src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Objects;
+
 public class WorkWithFile {
+    private static final String SUPPLY_TEXT = "supply";
+    private static final String BUY_TEXT = "buy";
+    private static final String RESULT_TEXT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] data = getData(fromFileName);
+        createReport(toFileName, data[0], data[1]);
+    }
+
+    public void createReport(String filename, int buy, int supply) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename))) {
+            StringBuilder reportBuilder = new StringBuilder();
+            String reportString = reportBuilder.append(SUPPLY_TEXT)
+                    .append(",")
+                    .append(supply)
+                    .append(System.lineSeparator())
+                    .append(BUY_TEXT)
+                    .append(",")
+                    .append(buy)
+                    .append(System.lineSeparator())
+                    .append(RESULT_TEXT)
+                    .append(",")
+                    .append(supply - buy)
+                    .toString();
+            writer.write(reportString);
```

Comment:
SRP violation - mixing of IO and business logic. consider refactor like in suggestion